x/bank/keeper/view.go

```diff
@@ -67,6 +68,32 @@ func (k BaseViewKeeper) GetAllBalances(ctx sdk.Context, addr sdk.AccAddress) sdk
 	return balances.Sort()
 }
 
+// GetAccountsBalances returns all the accounts balances from the store.
+func (k BaseViewKeeper) GetAccountsBalances(ctx sdk.Context) []types.Balance {
+	balances := make([]types.Balance, 0)
+	mapAddressToBalancesIdx := make(map[string]int)
+
+	k.IterateAllBalances(ctx, func(addr sdk.AccAddress, balance sdk.Coin) bool {
+		idx, ok := mapAddressToBalancesIdx[addr.String()]
+		if ok {
```

Comment:
why is it possible for the address to be iterated twice? The address is the key and the balance is the value right?

Reply:
the key is: `prefix + address + balance.Denom`

Reply:
oh I see :+1: that makes sense